mobile/src/main/java/org/openhab/habdroid/ui/AbstractBaseActivity.kt

```diff
@@ -134,25 +140,40 @@ abstract class AbstractBaseActivity : AppCompatActivity(), CoroutineScope {
     private fun timestampNeedsReauth(ts: Long) =
         ts == 0L || SystemClock.elapsedRealtime() - ts > AUTHENTICATION_VALIDITY_PERIOD
 
-    @CallSuper
-    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
-        super.onActivityResult(requestCode, resultCode, data)
-        if (requestCode == SCREEN_LOCK_REQUEST_CODE) {
-            if (resultCode == RESULT_OK) {
-                lastAuthenticationTimestamp = SystemClock.elapsedRealtime()
-            } else if (getPrefs().getScreenLockMode(this) == ScreenLockMode.KioskMode) {
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
-                    finishAndRemoveTask()
-                } else {
-                    finish()
-                }
-            }
+    private inner class AuthPrompt : BiometricPrompt.AuthenticationCallback() {
+        private val contentView = findViewById<View>(R.id.activity_content)
+        private val prompt = BiometricPrompt(this@AbstractBaseActivity, Dispatchers.Main.asExecutor(), this)
+
+        fun authenticate() {
+            val info = BiometricPrompt.PromptInfo.Builder()
+                .setTitle(getString(R.string.app_name))
+                .setDescription(getString(R.string.screen_lock_unlock_screen_description))
+                .setDeviceCredentialAllowed(true)
+                .build()
+            contentView.isInvisible = true
+            prompt.authenticate(info)
+        }
+
+        override fun onAuthenticationError(errorCode: Int, errString: CharSequence) {
+            super.onAuthenticationError(errorCode, errString)
+            finishAndRemoveTaskIfPossible()
```

Comment:
I don't know _why_ I used `finishAndRemoveTask()` previously and I think just finishing here is better.